src/cim/components/ClusterDeployment/ClusterDeploymentHostsSelectionAdvanced.tsx

```diff
@@ -0,0 +1,126 @@
+import { GridItem } from '@patternfly/react-core';
+import { useFormikContext } from 'formik';
+import React from 'react';
+import { LabelField } from '../../../common';
+import HostsTable from '../../../common/components/hosts/HostsTable';
+import {
+  cpuCoresColumn,
+  disksColumn,
+  hostnameColumn,
+  memoryColumn,
+} from '../../../common/components/hosts/tableUtils';
+import { AgentK8sResource } from '../../types';
+import { AgentTableEmptyState } from '../Agent/AgentTable';
+import { infraEnvColumn, statusColumn, useAgentsTable } from '../Agent/tableUtils';
+import { AGENT_LOCATION_LABEL_KEY } from '../common';
+import LocationsSelector from './LocationsSelector';
+import {
+  ClusterDeploymentHostsSelectionProps,
+  ClusterDeploymentHostsSelectionValues,
+  ClusterDeploymentHostsTablePropsActions,
+} from './types';
+
+type AgentsSelectionTableProps = {
+  matchingAgents?: AgentK8sResource[];
+  actions: ClusterDeploymentHostsTablePropsActions;
+};
+
+const AgentsSelectionTable: React.FC<AgentsSelectionTableProps> = ({
```

Comment:
AgentsSelectionTable is basically a custom formik field. Instead of using `useFormikContext` we can use `useField`

Reply:
done